src/components/Header/Header.tsx

```diff
@@ -0,0 +1,155 @@
+import {
+  memo, useCallback, useContext, useEffect, useMemo, useRef, useState,
+} from 'react';
+import cn from 'classnames';
+import * as todoService from '../../api/todos';
+import { StateContext, DispatchContext } from '../../store/store';
+import { ActionType } from '../../types/ActionType';
+import { ShowError } from '../../types/ShowErrors';
+import { USER_ID } from '../../types/constants';
+
+export const Header:React.FC = memo(() => {
+  const [newTodoTitle, setNewTodoTitle] = useState<string>('');
+
+  const { todos, loadingIDs } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+
+  const inputNewTitleRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputNewTitleRef.current && loadingIDs === null) {
+      inputNewTitleRef.current.focus();
+    }
+  }, [newTodoTitle, loadingIDs]);
+
+  const isAllCompleted = useMemo(
+    () => todos.every(todo => todo.completed),
+    [todos],
+  );
+
+  const isDisabled = useMemo(
+    () => !(loadingIDs === null),
+    [loadingIDs],
+  );
+
+  const isGreaterZero = useMemo(
```

Comment:
memoization also needs some resources so it's not effective to `useMemo` everywhere - [article](https://www.freecodecamp.org/news/memoization-in-javascript-and-react/#when-to-use-the-usememo-hook), in the article is the link to more info about  "When to useMemo and useCallback"

just FYI